common/debug-interface/src/lib.rs

```diff
@@ -49,18 +49,24 @@ impl NodeDebugClient {
         }
     }
 
-    fn client(&mut self) -> Result<NodeDebugInterfaceClient<tonic::transport::Channel>> {
-        let client = if let Some(client) = self.client.take() {
-            client
-        } else {
-            self.rt
-                .block_on(NodeDebugInterfaceClient::connect(self.addr.clone()))?
-        };
-        Ok(client)
-    }
+    fn client(
+        &mut self,
+    ) -> Result<(
+        &mut Runtime,
+        &mut NodeDebugInterfaceClient<tonic::transport::Channel>,
+    )> {
+        if self.client.is_none() {
+            self.client = Some(
+                self.rt
+                    .block_on(NodeDebugInterfaceClient::connect(self.addr.clone()))?,
+            );
+        }
 
-    fn set_client(&mut self, client: NodeDebugInterfaceClient<tonic::transport::Channel>) {
-        self.client = Some(client);
+        // client is guaranteed to be populated by the time we reach here
+        Ok((
+            &mut self.rt,
+            self.client.get_or_insert_with(|| unreachable!()),
```

Comment:
Is this really better than unwrap?

Reply:
The difference here is that `get` returns an &mut ref, though you're right its probably better to just use `as_mut().unwrap()`

Reply:
I think so. It has a simpler to understand meaning.